### ISSUE TITLE
Issues installing on OSX using ZSH

### ISSUE THREAD

**Issue by earlbalai** [dotnet/cli#1567]
The path to dotnet was not added to my zshrc automatically during installation. After manually adding it myself it works like a charm. 

os: 10.11
zsh version: zsh 5.0.8 (x86_64-apple-darwin15.0)
dotnet version: 1.0.0-beta-001517 

@blackdwarf

**Comment by agocke** [comment]
Maybe if we detect a ~/.zshrc we can print a message suggesting they add the os x `path_helper` call to their zshrc? I don't think we should change peoples' `.rc` files without approval.

**Comment by blackdwarf** [comment]
@agocke no, changing people's .rc files is out of question. I'm still confused about why this works on my machine, so repro time. :)

**Comment by brthor** [comment]
This shouldn't beed to be added to the .rc file right, we symlink to /usr/bin?

**Comment by blackdwarf** [comment]
Found this http://unix.stackexchange.com/questions/22979/path-helper-and-zsh. According to this, `path_helper` is not getting called in zsh. This would explain @earlbalai and @barambani issues with not having the tools in the $PATH. 

There are two ways to solve this:

1. We detect and modify the .zshrc file by `echo "export PATH=$PATH:/usr/local/share/dotnet >> .zshrc` or similar. This would solve the problem but is invasive.
2. We print out the warning that if you use zsh you need to add this to your path or invoked the `path_helper` from the profile.

**Comment by blackdwarf** [comment]
Of course, this still doesn't explain why is it working for me :(

**Comment by blackdwarf** [comment]
@earlbalai @barambani can you guys check if you have a file in `/etc/` which is called `zprofile` and does it contain anything?

**Comment by brthor** [comment]
@blackdwarf What if we symlink to /usr/bin again? Are the bugs preventing that still happening?

**Comment by blackdwarf** [comment]
@brthor no, but...seems *wrong* to do it. Also, would like to know if @earlbalai has El Capitan, 'cause it seems that the zprofile is something that El Capitan brings by default, which would explain why the installer "just worked" on my machine.

**Comment by barambani** [comment]
no `zprofile` in `/etc/` for me, but I have a file `zshenv` there that contains the lines

```
# system-wide environment settings for zsh(1)
if [ -x /usr/libexec/path_helper ]; then
	eval `/usr/libexec/path_helper -s`
fi```

I'm on 10.10.5 Yosemite

**Comment by blackdwarf** [comment]
@barambani interestingly enough, of El Capitan I **don't** have zshenv but I have zprofile and it works. Weird. 

I think the solution could be that we detect the presence of several of the zsh-specific files and then drop this file.

**Comment by barambani** [comment]
This thread might give some explanations 

http://www.zsh.org/mla/users/2015/msg00724.html

it also looks like that this changed only in El Capitan, so might be enough to support two variants

**Comment by xabikos** [comment]
I am in OSX El Capitan 10.11.3 and using zsh and faced the same problem. I had to execute in a terminal window the command "export PATH=$PATH:/usr/local/share/dotnet/bin" >> .zshrc as suggested above and problem was fixed.

**Comment by shawnrc** [comment]
Since the project requires users to have brew set up, why not create a symlink to the dotnet binary in `/usr/local/bin/`? Also doesn't require root privilege to do so.

**Comment by blackdwarf** [comment]
@shawnrc not quite sure what you mean? We do not have a brew formula (yet).

**Comment by shawnrc** [comment]
@blackdwarf On the OS X install instructions page (https://www.microsoft.com/net/core#macosx), the first thing they have users do is to install brew and pull down the latest version of openssl. If I remember correctly, the brew installer instructs users to add `/usr/local/bin/` to their PATH, so you might be able to take advantage of that.

**Comment by brthor** [comment]
Why `/usr/local/bin` over `/usr/bin` ? 

Is pkg considered machine wide or user specific?

**Comment by blackdwarf** [comment]
@shawnrc ah, I see what you mean. Yes, we might end up doing that if we detect `zsh` on the user's machine. 

But, my main thig is, `path_helper` is an OS-wide thing. I would like to know why is this not working for `zsh` and whether any solution we come up with is a workaround or permanent.

**Comment by blackdwarf** [comment]
@brthor PKG is machine wide.

**Comment by shawnrc** [comment]
@blackdwarf Fair enough, and figuring out how to get `path_helper` working properly on all shells is potentially the more sustainable solution, assuming you can't symlink to some directory that's more likely to already be part of users' PATHs. 

Crazy idea - perhaps a `brew`-installed `zsh` doesn't use the system zprofile at `/etc/zprofile`, and therefore doesn't make a call to `path_helper`?

**Comment by blackdwarf** [comment]
@shawnrc interesting idea. Will look into that.

**Comment by blackdwarf** [comment]
Guys, can you post your `.zshrc` and tell me if you are using oh-my-zsh?

**Comment by blackdwarf** [comment]
After some thinking about this, I think that this is really a problem of various different shell installers. We will have documentation to essentially have a workaround to add the symbolic link manually.

**Comment by piotrpMSFT** [comment]
Fixed by #3240
